src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -72,27 +104,54 @@ def write_to_tmp_file(content):
             tmp_file = tempfile.NamedTemporaryFile(delete=False)
             tmp_file.write(content.encode('utf-8'))
             tmp_file.flush()  # tmp_file must not be gc'ed
+            tmp_file.close()
             return tmp_file
 
+        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+        is_cephadm = orch_backend == 'cephadm'
+
         if module_name not in ['prometheus', 'alertmanager']:
-            raise DashboardException(f'Invalid module name {module_name}', component='prometheus')
+            raise DashboardException(f'Invalid module name {module_name}',
+                                     coFalsemponent='prometheus')
+
         user = None
         password = None
         cert_file = None
         pkey_file = None
         ca_cert_file = None
-        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
-        if orch_backend == 'cephadm':
-            cmd = {'prefix': f'orch {module_name} get-credentials'}
-            ret, out, _ = mgr.mon_command(cmd)
-            if ret == 0 and out is not None:
-                access_info = json.loads(out)
-                if access_info:
-                    user = access_info['user']
-                    password = access_info['password']
-                    ca_cert_file = write_to_tmp_file(access_info['certificate'])
-                    cert_file = write_to_tmp_file(mgr.get_localized_store("crt"))
-                    pkey_file = write_to_tmp_file(mgr.get_localized_store("key"))
+
+        if is_cephadm:
+            # Check cache first
+            cached_creds = self._get_cached_credentials(module_name)
+            if cached_creds:
+                return (cached_creds['user'], cached_creds['password'],
+                        cached_creds['ca_cert_file'], cached_creds['cert_file'],
+                        cached_creds['pkey_file'])
+
+            secure_monitoring_stack = mgr.get_module_option_ex('cephadm', 'secure_monitoring_stack')
+            if secure_monitoring_stack:
```

Comment:
similar to the previous comment, if you use `if not secure_monitoring_stack`: you can return earlier and void the indentation.